gcn_classic_text_to_json/notices/calet/__init__.py

```diff
@@ -0,0 +1,109 @@
+        "trigger_time": ["TRIGGER_DATE:", "TRIGGER_TIME:"],
+    },
+    "additional": {
+        "url": ("LC_URL:", "string"),
+        "rate_snr": ("SIGNIFICANCE:", "float"),
+        "foreground_duration": ("FOREGND_DUR:", "float"),
+        "background_duration": ("BACKGND_DUR1:", "float"),
+    },
+}
+
+
+def text_to_json_calet(data, record_number, notice_start_idx):
+    """Function calls text_to_json and then adds additional fields with cannot be dealt with by the general function.
+
+    Parameters
+    -----------
+    data: string
+        The text notice that is being parsed.
+    record_number: int
+        The current notice in the webpage being parsed.
+    notice_start_idx: int
+        The index at which the current text notice begins.
+
+    Returns
+    -------
+    dictionary
+        A dictionary compliant with the associated schema for the mission."""
+    output_dict = conversion.text_to_json(data, input, notice_start_idx)
+
+    output_dict["record_number"] = record_number
+    if record_number == 1:
+        output_dict["alert_type"] = "initial"
+    else:
+        output_dict["alert_type"] = "update"
+
+    output_dict["mission"] = "CALET"
+    output_dict["instrument"] = "GBM"
+    output_dict["trigger_type"] = "rate"
+
+    lon_lat_data = utils.find_string_between(
+        data, "SC_LON_LAT:", "\n", notice_start_idx
+    ).split(",")
+    lon, lat = lon_lat_data[0], lon_lat_data[1][:5]
+    output_dict["longitude"] = float(lon)
+    output_dict["latitude"] = float(lat)
+
+    energy_data = utils.find_string_between(
+        data, "ENERGY_BAND:", "\n", notice_start_idx
+    )
+    energy_low, energy_high = energy_data.split()[0].split("-")
+    output_dict["rate_energy_range"] = [int(energy_low), int(energy_high)]
+
+    detector_data = utils.find_string_between(
+        data, "TRIGGER_DET:", "\n", notice_start_idx
+    ).split()
+    detector_opts = ["on", "triggered"]
+    output_dict["detector_status"] = {
+        detector_data[-3][1:]: detector_opts[int(detector_data[0])],
+        detector_data[-2]: detector_opts[int(detector_data[1])],
+        detector_data[-1][:-1]: detector_opts[int(detector_data[2])],
+    }
+
+    return output_dict
+
+
+def create_all_calet_jsons():
+    """Creates a `calet_json` directory and fills it with the json for all CALET triggers."""
+    output_path = "./calet_jsons/"
+    if not os.path.exists(output_path):
+        os.makedirs(output_path)
+
+    link = "https://gcn.gsfc.nasa.gov/calet_triggers.html"
+    trigger_list = conversion.parse_trigger_ids(link)
+
+    for trig_num in trigger_list:
+        notice_link = f"https://gcn.gsfc.nasa.gov/other/{trig_num}.calet"
+        data = requests.get(notice_link).text
+
+        start_idx = 0
+        record_number = 1
+        while True:
+            output = text_to_json_calet(data, record_number, start_idx)
+
+            f = open(output_path + f"CALET_GBM_{trig_num}_{record_number}.json", "w")
+            json.dump(output, f)
```

Comment:
You are already using an f-string to format the file name, you can include the `output_path` 

```suggestion
            f = open(f"{output_path}CALET_GBM_{trig_num}_{record_number}.json", "w")
            json.dump(output, f)
```